src/main/scala/scalismo/faces/color/ColorDistribution.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+package scalismo.faces.color
+import scalismo.utils.Random
+
+import scala.collection.immutable.IndexedSeq
+
+trait ColorDistribution {
+
+  def sample(implicit rnd: Random): RGB
+
+  def normalizer: Double
+
+  def logNormalizer: Double
+
+  def evaluate(color: RGB): Double
+
+  def evaluateLog(color: RGB): Double
+}
+
+case class GaussDist(mean: RGB, sdev: RGB) extends ColorDistribution {
+  override def sample(implicit rnd: Random): RGB = {
+    val r = RGB(rnd.scalaRandom.nextGaussian(), rnd.scalaRandom.nextGaussian(), rnd.scalaRandom.nextGaussian())
```

Comment:
Implement a TruncatedGaussianColorDistribution instead (and call the method differently). I think that is a correct implementation for a GaussianColorDistribution, returning invalid RGB values would also be ugly.

Reply:
But it does not make sense to call it Gaussian because it has heavy peaks at the border where you clamp. Everything outside of this border is move onto the border.  I assume these peaks are not what you want. Instead of clamping you could simply regenerate a random number when the sampled value is outside of the color range on per color channel basis. I think this would directly lead to the correct truncated Gaussian which then corresponds to what you calculate in the evaluate function. But of course leading to a computational overhead.

Reply:
We have to be careful here, since the remaining implementation builds on a proper Gaussian distribution.
The handling of the RGB range is another topic. Here it is handled (there are several ways to handle it). We btw. have the same issue when sampling from our color morphable model...

Reply:
the draw sample method is btw at the moment only used for visualization

Reply:
I now removed the clamping from the GaussianColorDistribution and added it to where I apply it for visualization. This is not nice but consistent with the behaviour from the color part of the morphable model (no clamping when sampling). Like this the user can decide how to handle those value, to draw a new sample or to clamp it

Reply:
I like this solution that you moved the clamping. This makes this class consitent with the `sample` and `evaluate` function. That it leads to improper color values one can possibly infere from the name GaussianColorDistribution.